deepsensor/model/model.py

```diff
@@ -621,6 +621,160 @@ def unnormalise_pred_array(arr, **kwargs):
+
+        Returns:
+            :class:`~.model.pred.Prediction`):
+                A `dict`-like object mapping from target variable IDs to xarray or pandas objects
+                containing model predictions.
+                - If ``X_t`` is a pandas object, returns pandas objects
+                containing off-grid predictions.
+                - If ``X_t`` is an xarray object, returns xarray object
+                containing on-grid predictions.
+                - If ``n_samples`` == 0, returns only mean and std predictions.
+                - If ``n_samples`` > 0, returns mean, std and samples
+                predictions.
+
+        Raises:
+            ValueError
+                If ``X_t`` is not an xarray object and
+                ``resolution_factor`` is not 1 or ``ar_subsample_factor`` is
+                not 1.
+            ValueError
+                If ``X_t`` is not a pandas object and ``append_indexes`` is not
+                ``None``.
+            ValueError
+                If ``X_t`` is not an xarray, pandas or numpy object.
+            ValueError
+                If ``append_indexes`` are not all the same length as ``X_t``.
+        """
+
+        # Identify extent of original dataframe
+        preds = []
+        for task in tasks:
+            bbox = task['bbox']
+
+            # Determine X_t for the patched task in original coordinates.
+            x1 = xr.DataArray([bbox[0], bbox[1]], dims='x1', name='x1')
+            x2 = xr.DataArray([bbox[2], bbox[3]], dims='x2', name='x2')
+            bbox_norm = xr.Dataset(coords={'x1': x1, 'x2': x2})
+
+            bbox_unnorm = data_processor.unnormalise(bbox_norm)
+            unnorm_bbox_x1 = bbox_unnorm['x'].values.min(), bbox_unnorm['x'].values.max()
+            unnorm_bbox_x2 = bbox_unnorm['y'].values.min(), bbox_unnorm['y'].values.max()
+
+            task_X_t = X_t.sel(x = slice(unnorm_bbox_x1[0], unnorm_bbox_x1[1]),
+                                y = slice(unnorm_bbox_x2[0], unnorm_bbox_x2[1]))
+
+            # Patchwise prediction
+            pred = self.predict(task, task_X_t)
+            preds.append(pred)
+
+        # Produce a blank xarray to stitch patched predictions to.
+        pred_copy = copy.deepcopy(preds[0])
+
+        for var_name_copy, data_array_copy in pred_copy.items():
+
+            # set x and y coords
+            stitched_preds = xr.Dataset(coords={'x': X_t['x'], 'y': X_t['y']})
+
+            # Set time to same as patched prediction
+            stitched_preds['time'] = data_array_copy['time']
+
+            # set variable names to those in patched prediction, make values blank
+            for var_name_i in data_array_copy.data_vars:
+                stitched_preds[var_name_i] = data_array_copy[var_name_i]
+            stitched_preds.attrs.clear()
+            pred_copy[var_name_copy]= stitched_preds
+
+        # Stitch patchwise predictions
+        for pred in preds:
```

Comment:
Wondering, if xarray can be used instead of the nested loop directly with something like [combine_by_coords](https://docs.xarray.dev/en/latest/generated/xarray.combine_by_coords.html) ?